src/modules/roc_pipeline/target_stdio/roc_pipeline/parse_port.cpp

```diff
@@ -47,97 +47,140 @@ bool match_proto(PortType type, const char* str, PortProtocol& proto) {
     return false;
 }
 
-} // namespace
+PortProtocol parse_proto(const char* begin, const char* end, PortType type) {
+    char proto_buf[16] = {};
 
-bool parse_port(PortType type, const char* input, PortConfig& result) {
-    if (input == NULL) {
-        roc_log(LogError, "parse port: string is null");
-        return false;
+    if (size_t(end - begin) > sizeof(proto_buf) - 1) {
+        roc_log(LogError, "parse port: bad protocol: too long");
+        return Proto_None;
     }
 
-    const char* lcolon = strchr(input, ':');
-    const char* rcolon = strrchr(input, ':');
+    memcpy(proto_buf, begin, size_t(end - begin));
+    proto_buf[end - begin] = '\0';
 
-    if (!lcolon || !rcolon || lcolon == rcolon || lcolon == input || !rcolon[1]) {
-        roc_log(LogError,
-                "parse port: bad format: expected PROTO:ADDR:PORT or PROTO::PORT");
+    PortProtocol protocol = Proto_None;
+    if (!match_proto(type, proto_buf, protocol)) {
+        return Proto_None;
+    }
+
+    return protocol;
+}
+
+long parse_port_num(const char* port) {
```

Comment:
s/long/int/

The returned port is guaranteed to be in the correct range.

Reply:
well, internally we are working with the port as a long number why should we return int and explicitly cast long to int?

Reply:
We're casting port to int anyway. I'm just suggesting to make the cast earlier because in parse_port_num() we already checked that the port is in [0; 2^16) so the cast is valid.